ddp/ddp_pipeline/src/main/java/org/mskcc/cmo/ks/ddp/pipeline/util/DDPUtils.java

```diff
@@ -166,6 +174,43 @@ public static String resolvePatientCurrentAge(DDPCompositeRecord compositeRecord
         return anonymizePatientAge(age.intValue());
     }
 
+    public static String getAgeAtSeqDateLogging(String sampleId, String ageAtSeqDate, Long referenceInDays) {
+        StringBuilder builder = new StringBuilder();
+        builder.append("Sample '").append(sampleId).append("' resolves AGE_AT_SEQ_REPORTED_YEARS to null':  ")
+                .append("( AGE_AT_SEQ_REPORTED_YEARS=").append(ageAtSeqDate)
+                .append(", REFERENCE_DATE_DAYS=").append(referenceInDays)
+                .append(" )");
+        return builder.toString();
+    }
+
+    /**
+     * Resolve and anonymize patient age in years at date of sequencing. If either
+     * the patient date of birth or the date of sequencing is null this returns null.
+     *
+     * @param sampleId
+     * @param patientBirthDate String date of birth
+     * @return age in years at sequencing, null if we don't know patient date of birth or sequencing date
+     * @throws ParseException
+     */
+    public static String resolveAgeAtSeqDate(String sampleId, String patientBirthDate) throws ParseException {
+        Long birthDateInDays = getDateInDays(patientBirthDate);
+        Long referenceDateInDays = getDateInDays(sampleSeqDateMap.get(sampleId));
+        // if either date is null do not calculate age at sequencing date
+        Double age = (referenceDateInDays == null || birthDateInDays == null) ? null : (referenceDateInDays - birthDateInDays) / (DAYS_TO_YEARS_CONVERSION);
+        if (age == null) {
+            // log cases where AGE_AT_SEQ_REPORTED_YEARS is null because required values are null
+            String osMonthsLogMessage = getAgeAtSeqDateLogging(sampleId, String.valueOf(age), referenceDateInDays);
+            LOG.debug(osMonthsLogMessage);
+        }
```

Comment:
If a
`return null`
is added in the if body then we could change "else if" into plain "if" below and also drop the ternary operator at the end of this function (if we also put a `return null` at then end of the (age < 0) conditional block)